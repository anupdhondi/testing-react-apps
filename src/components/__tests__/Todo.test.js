import { render, screen, cleanup } from '@testing-library/react';
import Todo from '../Todo';

afterEach(() => {
  cleanup();
});

test('should render completed todo', () => {
  const todo = {
    id: 1,
    title: 'Maths',
    completed: true,
  };
  render(<Todo todo={todo} />);
  const todoElement = screen.getByTestId('todo-1');
  expect(todoElement).toBeInTheDocument();
  expect(todoElement).toHaveTextContent('Maths');
});

test('should render not completed todo', () => {
  const todo = {
    id: 2,
    title: 'Physics',
    completed: false,
  };
  render(<Todo todo={todo} />);
  const todoElement = screen.getByTestId('todo-2');
  expect(todoElement).toBeInTheDocument();
  expect(todoElement).toHaveTextContent('Physics');
  expect(todoElement).not.toContainHTML('<strike>');
});
