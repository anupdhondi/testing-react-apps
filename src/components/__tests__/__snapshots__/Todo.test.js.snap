// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot 1`] = `
<div
  data-testid="todo-1"
>
  <strike>
    <h1>
      todo - 
      Maths
    </h1>
  </strike>
</div>
`;
